src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore && target.hidden === false) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        const targetIndex = Animal.alive.findIndex((el) => el === target);
+
+        Animal.alive.splice(targetIndex, 1);
```

Comment:
[CODE PERFORMANCE] - Instead of using `findIndex` and `splice` to remove the dead animal from the `Animal.alive` array, which involves nesting loops, you should use the `filter` method to avoid performance issues. For example, after line 32, you can add a line to reassign `Animal.alive` to a filtered version of itself that only includes animals with health above 0.